common/ferrostar/src/navigation_controller/models.rs

```diff
@@ -3,9 +3,11 @@ use crate::models::{RouteStep, SpokenInstruction, UserLocation, VisualInstructio
 #[cfg(feature = "alloc")]
 use alloc::vec::Vec;
 use geo::LineString;
+#[cfg(feature = "wasm-bindgen")]
+use serde::{Deserialize, Serialize};
 
 /// A subset of state values that are used to show the user their current progress along the trip and it's components.
-#[derive(Debug, Clone, PartialEq)]
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
```

Comment:
I'm guessing this is at least one cause of the known limitation you mentioned with builds not working. You've unconditionally derived these traits, but the import is (correctly) conditional. You can use `cfg_attr` here like you do below :)

(btw props for getting this far with features! This is a pretty advanced part of cargo that many rust devs never really need to learn :D)